manager.go

```diff
@@ -1335,3 +1339,31 @@ func (m *jobManager) finishJob(name string) {
 
 	delete(m.muJob.running, name)
 }
+
+type requestsClient struct {
```

Comment:
I don't know why you have this.  Every place you call it you already had to be holding the manager lock.  So this doesn't protect anything